src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java

```diff
@@ -304,6 +312,39 @@ public NgxLearningPathDTO generateNgxPathRepresentation(@NotNull LearningPath le
         return this.learningPathNgxService.generateNgxPathRepresentation(learningPath);
     }
 
+    /**
+     * Get the navigation for the given learning path.
+     *
+     * @param learningPathId     the id of the learning path
+     * @param learningObjectId   the id of the relative learning object
+     * @param learningObjectType the type of the relative learning object
+     * @return the navigation
+     */
+    public LearningPathNavigationDTO getLearningPathNavigation(long learningPathId, @Nullable Long learningObjectId, @Nullable LearningObjectType learningObjectType) {
+        var learningPath = findWithCompetenciesAndLearningObjectsAndCompletedUsersById(learningPathId);
+        if (!userRepository.getUser().equals(learningPath.getUser())) {
+            throw new AccessForbiddenException("You are not allowed to access this learning path");
+        }
```

Comment:
duplicated code, I suggest to refactor this into a method with a proper name related to security checks and to call this from the resource so that it's 100% clear from the resource that security checks are executed